Sources/Widgets/Widgets3D/ShapeWidget/behavior.js

```diff
@@ -27,6 +27,13 @@ function makeBoundsFromPoints(point1, point2) {
 export default function widgetBehavior(publicAPI, model) {
   model.classHierarchy.push('vtkShapeWidgetProp');
 
+  model.keysDown = {};
+  model.modifierBehavior = model.factory.getReferenceByName('modifierBehavior');
+  model.resetAfterPointPlacement = model.factory.getReferenceByName(
+    'resetAfterPointPlacement'
+  );
+  macro.setGet(publicAPI, model, ['resetAfterPointPlacement']);
```

Comment:
@floryst I'm not very happy with this part. Do you have something better to suggest ?

To me, those properties are global (hence initializable in `vtkShapeWidget.newInstance`) but are used only by the behavior (i.e. view widget).

Maybe we should extend `methodsToLink` to propagate properties betwen Widget and ViewWidget in addition to the existing property propagation between ViewWidget and Representation.

WDYT ?

Reply:
Do you mean `resetAfterPointPlacement` is global in the sense that `widget.setResetAfterPointPlacement(true)` should affect all view widgets? If so, line 32 break that assumption because `model.resetAfterPointPlacement` does not refer to the property from the factory. In that case, I would opt to be explicit, and say `publicAPI.setResetAfterPointPlacement = model.factory.setResetAfterPointPlacement`, and the same for the getter.

If you are only interested in passing initialization, then what you have should be fine for now. With regard to the `methodsToLink` extension idea, I'm not too much a fan of that either. I would much rather be explicit in initialization. 

Here's an idea: have ShapeWidget override `getWidgetForView`, like so:

```js
const superClass = { ...publicAPI };
publicAPI.getWidgetForView = (props) => {
  return superClass.getWidgetForView({
    ...props,
    initialValues: {
      modifierBehavior: model.modifierBehavior,
      resetAfterPointPlacement: model.resetAfterPointPlacement,
      ...props.initialValues,
    },
  });
};
```

You'll still need to have `macro.setGet(...)` in the widgetBehavior, but that's a consequence of how restricted our factory creation method (getWidgetForView) is.

Reply:
Thanks for your answer and ideas. 
> Do you mean resetAfterPointPlacement is global in the sense that widget.setResetAfterPointPlacement(true) should affect all view widgets?

Yes, this is what I want.
I like your solution `publicAPI.setResetAfterPointPlacement = model.factory.setResetAfterPointPlacement` because the property does not have to be stored twice.